source/common/http/http1/conn_pool.cc

```diff
@@ -203,7 +203,9 @@ void ConnPoolImpl::onResponseComplete(ActiveClient& client) {
   if (!client.stream_wrapper_->encode_complete_) {
     ENVOY_CONN_LOG(debug, "response before request complete", *client.codec_client_);
     onDownstreamReset(client);
-  } else if (client.stream_wrapper_->saw_close_header_ || client.codec_client_->remoteClosed()) {
+  } else if (client.stream_wrapper_->saw_close_header_ || client.codec_client_->remoteClosed() ||
+             (client.codec_client_->protocol() == Protocol::Http10 &&
+              !client.stream_wrapper_->saw_keep_alive_header_)) {
     ENVOY_CONN_LOG(debug, "saw upstream connection: close", *client.codec_client_);
```

Comment:
Maybe rephrase this log?

Reply:
Even before it was conflating RemoteClose.  Changed to:

 ENVOY_CONN_LOG(debug, "saw upstream close connection", *client.codec_client_);